Source/WTF/wtf/PlatformHave.h

```diff
@@ -1488,3 +1488,11 @@
 #if (PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 140000)
 #define HAVE_NSWINDOW_SNAPSHOT_READINESS_HANDLER 1
 #endif
+
+#if (PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 140000) \
+    || (PLATFORM(MACCATALYST) && __IPHONE_OS_VERSION_MIN_REQUIRED >= 170000) \
```

Comment:
Doesn't `PLATFORM(MAC)` include macCatalyst?

Also, what about visionOS?

Reply:
I believe `PLATFORM(IOS_FAMILY)` includes macCatalyst, but plain `PLATFORM(MAC)` does not. I'm not sure if `OS(VISION)` is part of `IOS_FAMILY` yet.

Reply:
I checked the headers yesterday, and I believe that both do. But it's complicated, and I could easily get it wrong without testing.

Reply:
Yes, I believe PLATFORM(IOS_FAMILY) includes macCatalyst, but PLATFORM(MAC) does not. This is not required for visionOS at the moment, I believe.

Reply:
> Yes, I believe PLATFORM(IOS_FAMILY) includes macCatalyst, but PLATFORM(MAC) does not.

Indeed.